de1plus/plugins/webhook/plugin.tcl

```diff
@@ -0,0 +1,129 @@
+        add_de1_widget $page_name entry $x [expr $y + 60]  {
+            bind $widget <Return> { say [translate {save}] $::settings(sound_button_in); borg toast [translate "Saved"]; save_plugin_settings webhook; hide_android_keyboard}
+            bind $widget <Leave> hide_android_keyboard
+        } -width [expr {int(38 * $::globals(entry_length_multiplier))}] -font Helv_8  -borderwidth 1 -bg #fbfaff  -foreground #4e85f4 -textvariable "$var" -relief flat  -highlightthickness 1 -highlightcolor #000000
+    }
+
+    proc post_shot_data {} {
+        variable settings
+
+        msg "triggering webhook"
+        borg toast [translate "Triggering Webhook"]
+
+        http::register https 443 [list ::tls::socket -servername $settings(webhook_domain)]
+
+        # Craft HTTP Body
+        set espresso_data [format_espresso_for_history]
+        set clock [clock seconds]
+        set boundary "--------$clock"
+        set type "multipart/form-data, charset=utf-8, boundary=$boundary"
+        set content [encoding convertto utf-8 $espresso_data]
+        set contentHeader "Content-Disposition: form-data; name=\"file\"; filename=\"file.shot\"\r\nContent-Type: application/octet-stream\r\n"
+        set body "--$boundary\r\n$contentHeader\r\n$content\r\n--$boundary--\r\n"
+
+        # Build HMAC signature
+        # HMAC Signature = SHA256(secret, endpoint_path + "\n" + clock + "\n" + body)
+        set message_for_hmac "$settings(webhook_endpoint)\n$clock\n$body"
+        set signature [sha2::hmac $settings(webhook_secret_key) $message_for_hmac]
+        set headerl [list \
+            Authorization "$clock!$signature" \
+            Content-Type "multipart/form-data; boundary=$boundary"]
+
+        if {[catch {
+            # HTTPS only for now
+            set url "https://$settings(webhook_domain)$settings(webhook_endpoint)"
+            set token [http::geturl $url -headers $headerl -method POST -type $type -query $body -timeout 30000]
+            set status [http::status $token]
+            set answer [http::data $token]
+            set returncode [http::ncode $token]
+            set returnfullcode [http::code $token]
+            msg $token
+            msg "status: $status"
+            # msg "answer: $answer"
+        } err] != 0} {
+            msg "Could not upload shot to webhook! $err"
+            borg toast [translate "Webhook failed!"]
+            catch { http::cleanup $token }
+            return
+        }
+
+        http::cleanup $token
+        if {$returncode != 200} {
+            msg "Webhook failed: $returnfullcode"
+            borg toast "Webhook failed!"
+            return
+        }
+
+        msg "Webhook succeeded!"
+        borg toast "Webhook succeeded!"
+    }
+
+    proc async_dispatch {old new} {
+        after [expr 1000 * $::plugins::webhook::settings(webhook_trigger_after)] ::plugins::webhook::post_shot_data
+    }
+
+    proc msg { msg } {
```

Comment:
You can go straight to `::logging::default_logger`

https://github.com/decentespresso/de1app/blob/main/de1plus/logging.tcl#L6